src/cclib/writer/cjsonwriter.py

```diff
@@ -17,97 +17,424 @@
+        "mocoeffs":       'coeffs',
+        "moenergies":     'energies',
+        "moments":        'total dipole moment',
+        "mosyms":         'symmetry',
+        "mpenergies":     'moller plesset',
+        "mult":           'multiplicity',
+        "natom":          'number of atoms',
+        "nbasis":         'basis number',
+        "nmo":            'MO number',
+        "nocoeffs":       'TBD',
+        "nooccnos":       'TBD',
+        "optdone":        'done',
+        "optstatus":      'status',
+        "scancoords":     'step geometry',
+        "scanenergies":   'PES energies',
+        "scannames":      'variable names',
+        "scanparm":       'PES parameter values',
+        "scfenergies":    'energies',
+        "scftargets":     'targets',
+        "scfvalues":      'values',
+        "temperature":    'temperature',
+        "vibanharms":     'anharmonicity constants',
+        "vibdisps":       'displacement',
+        "vibfreqs":       'frequencies',
+        "vibirs":         'IR',
+        "vibramans":      'raman',
+        "vibsyms":        'symmetry',
+    }
 
-    def __init__(self, ccdata, *args, **kwargs):
+    def __init__(self, ccdata, cjsonterse=False, *args, **kwargs):
         """Initialize the chemical JSON writer object.
 
         Inputs:
           ccdata - An instance of ccData, parsed from a logfile.
         """
 
         # Call the __init__ method of the superclass
-        super(CJSON, self).__init__(ccdata, *args, **kwargs)
+        super(CJSON, self).__init__(ccdata, cjsonterse=cjsonterse, *args, **kwargs)
 
-        self.generate_repr()
+    def pathname(self, path):
+        """
+        This function is OS independent and returns the file name irrespective of
+        the file path containing forward slash or backward slash - which is valid
+        in Windows
+        """
+        head, tail = ntpath.split(path)
+        return tail or ntpath.basename(head)
 
     def generate_repr(self):
-        """Generate the CJSON representation of the logfile data."""
+        """Generate the CJSON representation of the logfile data"""
 
         cjson_dict = dict()
+        
+        # Need to decide on a number format
         cjson_dict['chemical json'] = 0
-
         if self.jobfilename is not None:
-            cjson_dict['name'] = os.path.splitext(self.jobfilename)[0]
+            cjson_dict['name'] = self.pathname(os.path.splitext(self.jobfilename)[0])
 
         # These are properties that can be collected using Open Babel.
-
         if has_openbabel:
             cjson_dict['smiles'] = self.pbmol.write('smiles')
             cjson_dict['inchi'] = self.pbmol.write('inchi')
             cjson_dict['inchikey'] = self.pbmol.write('inchikey')
             cjson_dict['formula'] = self.pbmol.formula
+            
+        # Incorporate Unit Cell into the chemical JSON
+
+        # Helpers functions which use properties provided by cclib
+        self.generate_properties(cjson_dict)
```

Comment:
I'm wondering whether this would be easier to maintain if we could separate the structure of the CJSON from the actual generation/writing. Can you think of a data structure that we could use to represent that structure on it's own? Not necessarily to do now, but would be nice.